flake.nix

```diff
@@ -77,6 +77,14 @@
           buildFeatures = [ "packaging" ];
           cargoBuildFlags = [ "--bin" "jj" ]; # don't build and install the fake editors
           useNextest = true;
+
+          checkFlags = [
+            # https://github.com/martinvonz/jj/issues/4784
+            # seems to fail reliably under Nix/nix-shell only, but otherwise
+            # is an unknown anomaly?
+            "--skip" "test_shallow_commits_lack_parents"
```

Comment:
The nextest native syntax would involve `--filterset`, and AIUI you need a `--` for `--skip` – does this actually work to skip it?

Reply:
I'm aware of the filterset expression, but I chose this in part because I think `--skip` reads much more naturally (part of the reason it was presumably added back to nextest, along with the fact it's backwards compatible with older versions; when I originally tried diagnosing this I had to spend 5 minutes figuring out why `--skip` was reworked and the right way to skip tests, which involved [a nice bug report](https://github.com/nextest-rs/nextest/issues/1816).) Regarding the `--` yes, it's because the actual Nix harness around `cargo test` adds it for you. I've never seen `--skip` used in any other way in any Rust+Nix project FWIW, so I've assumed this was always the default behavior.